core/src/main/java/com/graphhopper/routing/weighting/CurvatureWeighting.java

```diff
@@ -47,7 +47,8 @@ public CurvatureWeighting(FlagEncoder flagEncoder, PMap pMap, TurnCostProvider t
         curvatureEnc = flagEncoder.getDecimalEncodedValue(EncodingManager.getKey(flagEncoder, "curvature"));
         avSpeedEnc = flagEncoder.getDecimalEncodedValue(EncodingManager.getKey(flagEncoder, "average_speed"));
         double minBendiness = 1; // see correctErrors
-        minFactor = minBendiness / Math.log(flagEncoder.getMaxSpeed()) / PriorityCode.getValue(BEST.getValue());
+        double maxSpeed = avSpeedEnc.getRealMax() < 0 ? avSpeedEnc.getMaxDecimal() : avSpeedEnc.getRealMax();
```

Comment:
This is an important catch: So far we do not enforce the weightings to be created after the graph import is over, so we cannot guarantee that the maximum value is already determined. There is even GraphHopper#checkProfileConsistency which explicitly creates weightings before the import.
A practical solution to this problem seems to check whether the max value has been touched yet (check if it is still negative) and if not just fall back to the maximum storable value.